tests/e2e/specs/api-cache.test.js

```diff
@@ -0,0 +1,124 @@
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import {
+	activatePlugin,
+	deactivatePlugin,
+	visitAdminPage,
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	safeLoginUser,
+	setSearchConsoleProperty,
+	setSiteVerification,
+	testSiteNotification,
+	useRequestInterception,
+	wpApiFetch,
+} from '../utils';
+
+const goToSiteKitDashboard = async () => {
+	await visitAdminPage( 'admin.php', 'page=googlesitekit-dashboard' );
+};
+
+describe( 'API cache', () => {
+	beforeAll( async () => {
+		await page.setRequestInterception( true );
+		useRequestInterception( ( request ) => {
+			const url = request.url();
+			if ( url.match( 'search-console/data/searchanalytics' ) ) {
+				request.respond( { status: 200, body: '[]' } );
+			} else if ( url.match( 'pagespeed-insights/data/pagespeed' ) ) {
+				request.respond( { status: 200, body: '{}' } );
+			} else if ( url.match( 'user/data/survey-timeouts' ) ) {
+				request.respond( { status: 200, body: '[]' } );
+			} else {
+				request.continue();
+			}
+		} );
+
+		await activatePlugin( 'e2e-tests-proxy-auth-plugin' );
+		await setSiteVerification();
+		await setSearchConsoleProperty();
+	} );
+
+	afterAll( async () => {
+		await deactivatePlugin( 'e2e-tests-proxy-auth-plugin' );
+	} );
+
+	it( 'isolate client storage between users', async () => {
+		const firstTestNotification = { ...testSiteNotification };
+		const secondTestNotification = {
+			...testSiteNotification,
+			id: 'test-notification-2',
+			title: 'Test notification title 2',
+			dismissLabel: 'test dismiss site notification 2',
+		};
+
+		// create dummy first notification
+		await wpApiFetch( {
+			path: 'google-site-kit/v1/e2e/core/site/notifications',
+			method: 'post',
+			data: firstTestNotification,
+		} );
+
+		await goToSiteKitDashboard();
+
+		await page.waitForSelector(
+			`#${ firstTestNotification.id }.googlesitekit-publisher-win--is-open`
+		);
+
+		await expect( page ).toClick(
+			'.googlesitekit-publisher-win .mdc-button span',
+			{
+				text: firstTestNotification.dismissLabel,
+			}
+		);
+
+		// create dummy second notification
+		await wpApiFetch( {
+			path: 'google-site-kit/v1/e2e/core/site/notifications',
+			method: 'post',
+			data: secondTestNotification,
+		} );
+
+		// delete all cookies
+		await page._client.send( 'Network.clearBrowserCookies' );
```

Comment:
Interesting! It's probably better to avoid making direct calls (to private methods?) like this and instead use APIs that Puppeteer provides. E.g. https://github.com/google/site-kit-wp/blob/1bc31f1a4882b8db8d1f187d51be5ac618b845d7/tests/e2e/utils/logout-user.js#L10-L13

We could use this utility, but I'd prefer to make it more explicit that we explicitly want to delete the login cookies. How about we extract that part here to a new utility which is explicit about its behavior and then we can use that in this test?

Reply:
Makes sense, I created a new `deleteAuthCookie` utility function.